backend/src/flowable/listeners/flowable-event.listener.ts

```
@@ -0,0 +1,479 @@
+            FlowableEventListener.name,
+          );
+        }
+
+        const identityLinks = await this.flowableService.getTaskIdentityLinks(flowableTask.id);
+        const hasCandidateGroup = identityLinks.some((link: any) => link.type === 'candidate' && link.group);
+
+        if (!hasCandidateGroup && event.candidateGroup) {
+          await this.flowableService.assignTaskToCandidateGroup(flowableTask.id, event.candidateGroup);
+          this.logger.log(
+            `Assigned candidate group ${event.candidateGroup} to Flowable task ${flowableTask.id}`,
+            FlowableEventListener.name,
+          );
+        }
+      } else {
+        this.logger.warn(
+          `No matching BPMN task found for PostgreSQL task ${event.taskId} (${event.taskName}) in process ${processInstance.id}`,
+          FlowableEventListener.name,
+        );
+
+        flowableTasks.forEach((t: any) => {
+          this.logger.warn(`Available task: id=${t.id}, name="${t.name}", key="${t.taskDefinitionKey}"`, FlowableEventListener.name);
+        });
+      }
+    } catch (error) {
+      this.logger.error(`Failed to sync task ${event.taskId} with Flowable: ${error.message}`, error.stack, FlowableEventListener.name);
+    }
+  }
+
+  private getTaskDefinitionKey(taskName: string): string | null {
+    const mappings: Record<string, string> = {
+      'Approve Case Creation': 'approveCaseCreation',
+      'Investigate Case': 'investigateCase',
+      'Investigate case': 'investigateCase',
+      'Approve case closure': 'supervisorApproval',
+    };
+
+    return mappings[taskName] || null;
+  }
+
+  @OnEvent('task.status.changed')
+  async handleTaskStatusChanged(event: TaskStatusChangedEvent) {
+    try {
+      const processInstance = await this.flowableService.getProcessInstanceByBusinessKey(event.caseId);
+
+      if (!processInstance) {
+        this.logger.warn(`No Flowable process found for case ${event.caseId}`, FlowableEventListener.name);
+        return;
+      }
```

Should return an exception for process not found to provide meaningful response.